drivers/usb/device/usb_dc_native_posix.c

```diff
@@ -570,15 +572,20 @@ int handle_usb_data(struct usbip_header *hdr)
 
 		usbip_skip_setup();
 
-		LOG_DBG("Send %u bytes", usbip_ctrl.in_ep_ctrl[ep_idx].buf_len);
+		LOG_DBG("Send %u bytes", buf_len);
 
 		/* Send queued data */
-		usbip_send_common(ep, usbip_ctrl.in_ep_ctrl[ep_idx].buf_len);
-		usbip_send(ep, usbip_ctrl.in_ep_ctrl[ep_idx].buf,
-			   usbip_ctrl.in_ep_ctrl[ep_idx].buf_len);
+		if (!usbip_send_common(ep, buf_len)) {
+			return -EIO;
+		}
+
+		if (usbip_send(ep, usbip_ctrl.in_ep_ctrl[ep_idx].buf,
```

Comment:
I would consider adding `u8_t *buf = usbip_ctrl.in_ep_ctrl[ep_idx].buf;`

Reply:
done this way